### PR TITLE
Prevent PDF generation from getting license.png from template books (BL-4290)

### PR DESCRIPTION
**Pull request by andrew-polk**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/1512)

### CODE REVIEW

**Comment by StephenMcConnel** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1512#-:-KdCjfuWkpo1O4HTfcOl:bnfp4nl)*

File: src/BloomExe/ImageProcessing/ImageServer.cs [hunk 1]
```diff
@@ -74,7 +74,10 @@ protected override bool ProcessRequest(IRequestInfo info)
 			{
 				var fileName = Path.GetFileName(imageFile);
 				var sourceDir = FileLocator.GetDirectoryDistributedWithApplication(BloomFileLocator.BrowserRoot);
-				imageFile = Directory.EnumerateFiles(sourceDir, fileName, SearchOption.AllDirectories).FirstOrDefault();
+				
+				// Without filtering out the templates directory, we were picking up the license.png from template and shell books.
+				// This resulted in the wrong cc license icon when publishing to PDF. (BL-4290)
+				imageFile = Directory.EnumerateFiles(sourceDir, fileName, SearchOption.AllDirectories).FirstOrDefault(f => !f.Contains("templates"));
```
Comment (hatton) [inline src/BloomExe/ImageProcessing/ImageServer.cs]: Do we ever display books from the templates directory? If so, would this prevent some images from being retrieved/displayed?

**Comment by JohnThomson** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomExe/ImageProcessing/ImageServer.cs, line 80 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1512#-KdCi-mz9LVpJ6HylISH-r1-80:-KdQxKXulk65hB98m3uF:b-kmhfy7) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/26aca230cb454e60b81d7591d5272f4912cd2037/src/BloomExe/ImageProcessing/ImageServer.cs#L80)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

Do we ever display books from the templates directory? If so, would this prevent some images from being retrieved/displayed?
</blockquote></details>

There's also the (somewhat remote?) possibility that someone might be making a collection of templates (or even a book of some kind of pictures that are templates) and have that word in the path to the book folder itself.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1512)*

File: src/BloomExe/ImageProcessing/ImageServer.cs (hunk 1 above)
Comment (andrew-polk) [inline src/BloomExe/ImageProcessing/ImageServer.cs]: My testing shows any retrievals of assets from the templates directories do not go through this path.

**Comment by andrew-polk** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomExe/ImageProcessing/ImageServer.cs, line 80 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1512#-KdCi-mz9LVpJ6HylISH-r1-80:-KdR5oWat5HXKW-kkD11:bo7au84) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/26aca230cb454e60b81d7591d5272f4912cd2037/src/BloomExe/ImageProcessing/ImageServer.cs#L80)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

My testing shows any retrievals of assets from the templates directories do not go through this path.
</blockquote></details>

Regarding John Thomson's observation, that is actually fine in this case as we don't want it to be found. We want to return false here which will then correctly find the file later.

That being said, I'm not arguing for my approach. I was somewhat uneasy with it myself and am open to ideas of how to improve it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1512)*

**Comment by andrew-polk** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomExe/ImageProcessing/ImageServer.cs, line 80 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1512#-KdCi-mz9LVpJ6HylISH-r1-80:-KdR6o8DIO3LVIb6ziJy:b-a2cvup) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/26aca230cb454e60b81d7591d5272f4912cd2037/src/BloomExe/ImageProcessing/ImageServer.cs#L80)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Regarding John Thomson's observation, that is actually fine in this case as we don't want it to be found. We want to return false here which will then correctly find the file later.

That being said, I'm not arguing for my approach. I was somewhat uneasy with it myself and am open to ideas of how to improve it.
</blockquote></details>

This is only searching for files within our browser root directory.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1512)*

**Comment by JohnThomson** [comment]
I don't feel I fully understand this, but you answered the one question I had. Don't wait for me to merge it.

**Comment by andrew-polk** [comment]
Taking a different approach

**Pull request closed without merging**